Amazon.Ion.ObjectMapper/IonObjectSerializer.cs

```diff
@@ -73,6 +82,15 @@ public override object Deserialize(IIonReader reader)
 
         public override void Serialize(IIonWriter writer, object item)
         {
+            var customObjectSerializer = this.GetCustomObjectSerializer();
```

Comment:
I feel like this becomes simpler by using the `TryCustomObjectSerialize` paradigm instead.

Reply:
Done.